frontend/src/components/CourseModal/CourseModal.tsx

```diff
@@ -181,6 +183,22 @@ function CourseModal() {
       setHistory([listingFromQuery]);
     });
   }, [history.length, searchParams, requestSeasons, courses]);
+  useEffect(() => {
+    const crossSections: CourseModalHeaderData[] = [];
+    const courseModal = searchParams.get('course-modal');
+    if (!courseModal) return;
+    const seasonCode = courseModal.split('-')[0] as Season;
+    void requestSeasons([seasonCode]).then(() => {
+      courses[seasonCode]?.forEach((course: CourseModalHeaderData) => {
+        if (course.course_code === listing?.course_code)
+          crossSections.push(course);
+      });
+      setSections(
+        crossSections.sort((a, b) => a.section.localeCompare(b.section)),
+      );
+    });
+  }, [listing, courses, searchParams, requestSeasons]);
```

Comment:
also I think here we can make a new gql query instead of requesting the whole catalog right @Josh-Cena